### PR TITLE
Add JAX gradient checks for SBML testsuite

### PR DESCRIPTION
**Pull request by FFroehlich**
## Summary
- extend SBML semantic tests with JAX gradient checks
- compile SBML models for JAX and compare sensitivities
- tolerate JAX backend limitations by skipping models with event assignments
- adapt tolerances for JAX simulations
- fix failing sensitivity check by enabling sensitivities during simulation
- ensure model parameters request sensitivities for selected SBML cases

## Testing
- `pre-commit run --files tests/sbml/testSBMLSuite.py .github/workflows/test_sbml_semantic_test_suite_jax.yml`
- `pytest tests/benchmark_models/test_petab_benchmark.py -k Boehm_JProteomeRes2014 -q`
- `pytest tests/benchmark_models/test_petab_benchmark_jax.py -k Boehm_JProteomeRes2014 -q`
- `pytest tests/sbml/testSBMLSuite.py::test_sbml_testsuite_case[00783] -q`

------
https://chatgpt.com/codex/tasks/task_b_685d259297a4832bb74795ffd6214f34

### CODE REVIEW

**Review by dweindl — APPROVED** [review]
Great.

`sensitivity_check_cases` are those for which finite differences have been manually tested. Would be great to check gradients for all models (later).
